compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -556,73 +586,123 @@ private void defineReferencedClassFields(BLangClassDefinition classDefinition, S
             fieldNames.put(fieldVariable.name.value, fieldVariable);
         }
 
-        classDefinition.referencedFields = classDefinition.typeRefs.stream().flatMap(typeRef -> {
+        List<BLangSimpleVariable> referencedFields = new ArrayList<>();
```

Comment:
This is not about this PR but I think we should consider refactoring this method, maybe into a some sort of method object.

Reply:
We also need to revisit the field/method population logic for intersection types (https://github.com/ballerina-platform/ballerina-lang/issues/27948), which will probably help clean this up a bit too.